File: api_test.py
import json
import urllib.request
import urllib.error
import sys

def test_api(url):
    print(f"Testing {url}...")
    data = json.dumps({
        'uname': 'newtester1', 
        'email': 'newtest1@example.com', 
        'password': 'password123'
    }).encode('utf-8')
    
    req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'})
    try:
        res = urllib.request.urlopen(req)
        print(f"Success! Response: {res.read().decode('utf-8')}")
    except urllib.error.HTTPError as e:
        print(f"HTTP Error: {e.code} - {e.read().decode('utf-8')}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    if len(sys.argv) > 1:
        test_api(sys.argv[1])
    else:
        test_api('http://localhost:8000/api/auth/register/')
        test_api('https://moonflex.onrender.com/api/auth/register/')

File: api_test.js
const http = require('http');
const https = require('https');

function testAPI(urlStr) {
    console.log(`Testing ${urlStr}...`);
    const data = JSON.stringify({
        uname: 'test_node_user',
        email: 'test_node@example.com',
        password: 'password123'
    });

    const url = new URL(urlStr);
    const options = {
        hostname: url.hostname,
        port: url.port || (url.protocol === 'https:' ? 443 : 80),
        path: url.pathname,
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': data.length
        }
    };

    const req = (url.protocol === 'https:' ? https : http).request(options, (res) => {
        let body = '';
        res.on('data', (chunk) => body += chunk);
        res.on('end', () => {
            console.log(`Status: ${res.statusCode}`);
            console.log(`Response: ${body}`);
        });
    });

    req.on('error', (e) => {
        console.error(`Problem with request to ${urlStr}: ${e.message}`);
    });

    req.write(data);
    req.end();
}

testAPI('http://localhost:8000/api/auth/register/');
testAPI('https://moonflex.onrender.com/api/auth/register/');
